public/locations-list-map.js

```diff
@@ -768,23 +770,32 @@ function centerMapToMarkersNearUser() {
 function fitMapToMarkersNearBounds(bounds) {
   // get center of bounding box and use it to sort markers by distance
   let center = bounds.getCenter();
-  const markersByDistance = getMarkersByDistanceFrom(center.lat(), center.lng());
-  
+  const markersByDistance = getMarkersByDistanceFrom(center.lat(), center.lng(), 3);
+
+  let hasMarker = false;
+
   // extend bounds to fit closest three markers
   [0,1,2].forEach((i) => {
```

Comment:
Now you can just do markersByDistance.forEach, and don't need the if(marker) test.